docs/design/adrs/2024-05-02-enable-project-versioning

```diff
@@ -0,0 +1,63 @@
+
+## Decision Drivers
+
+* Minimal process (it should be straightforward to publish a release)
+* Small time overhead (steps should be automated where possible)
+* Low maintenance solution
+
+## Considered Options
+
+* GitHub Releases
+
+## Decision Outcome
+
+TBD
+
+## Pros and Cons of the Options
+
+### GitHub Releases
+
+GitHub Releases is a feature which can be enabled directly in the repository, and allows you to create releases with release notes, @mentions of contributors, and link assets such as binaries.
+
+Pros:
+
+* **Supported solution**: GitHub Releases is built-in to GitHub repositories, so is well supported and low overhead to enable. It is used by many product teams, e.g. [vscode](https://github.com/microsoft/vscode), [semantic-kernel](https://github.com/microsoft/semantic-kernel), [WSL](https://github.com/microsoft/WSL), etc.
+
+* **Easy publishing**: You can create, edit and delete releases using the browser or Releases API, which makes it simple to use.
+
+* **Built-in features**: You can compare releases from the repository, see [Comparing releases](https://docs.github.com/en/repositories/releasing-projects-on-github/comparing-releases). You can also see the latest release from the repository homepage.
+
+* **Automation**: There is support to automatically generate [release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes) based on changes since the previous release, it also includes who has contributed.
+
+* **Simple**: There are extensive [docs](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases).
+
+Cons:
+
+* **Automatic releases**: Whilst it is straight-forward to publish a release manually, there are extra steps required if we would want to have automatic releases.
+    - This [GitHub Blog post](https://github.blog/2021-12-16-5-automations-every-developer-should-be-running/#5-automate-your-releases-and-release-notes-with-github-actions) shares how you can use GitHub Actions to run a workflow which will create a release.
+    - If we were to automate the release, we would need to make decisions on the release schedule or trigger, for example, should a new release be created each time a change is merged to `main`? Or should it be done on a schedule such as daily, weekly, etc?
+    - If we had an automatic release, how would we deal with cases such as major version changes and making sure breaking changes didn't get included in a minor version release?
+* **Manual releases**: If we were to do manual releases, this would be a responsibility that would need to be assigned, for example to a Project Maintainer.
+
+## Proposal
+
+Based on the current activity in the project, we could:
+
+- Elect a Release Driver who will manually go through the steps of creating a release, this could be done during a call with the broader team and should take <30 mins.
```

Comment:
We are using a concept called a "Sprint Champion" #417 - maybe it should be the responsibility of that role to wrap up a release. I think the devil is in the detail about "how we release". One of the big things is that we have a "deploy to azure" button on the readme and published containers that are used with that code. I think we want that to have that "always work" and have less updates. Maybe we could add an interim branch "development" where we ship to main once a week? These are just ideas - could we explore that in the design?

Reply:
Yes! Will do a little more investigating and update the ADR

Reply:
@ross-p-smith I have updated the proposed changes, making it the responsibility of the Sprint Champion to review the releases. I've also updated the suggestion for "how" based on input from Bhavana - will expand on my reasoning below.